osu.Game/Overlays/Profile/Sections/PaginatedContainer.cs

```diff
@@ -97,16 +87,133 @@ private void onUserChanged(ValueChangedEvent<User> e)
+                        content.FadeOut(duration, Easing.OutQuint);
+                    }
+                    else
+                    {
+                        loading.FadeOut(duration, Easing.OutQuint);
+                        content.FadeIn(duration, Easing.OutQuint);
+                    }
+                }
+                get => isLoading;
+            }
+
+            public ShowMoreButton()
+            {
+                Anchor = Anchor.TopCentre;
+                Origin = Anchor.TopCentre;
+                Masking = true;
+                Size = new Vector2(140, 30);
+                Children = new Drawable[]
+                {
+                    background = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                    },
+                    content = new FillFlowContainer
+                    {
+                        Alpha = 0,
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        AutoSizeAxes = Axes.Both,
+                        Direction = FillDirection.Horizontal,
+                        Spacing = new Vector2(7),
+                        Children = new Drawable[]
+                        {
+                            new ChevronIcon(),
+                            new OsuSpriteText
+                            {
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Font = OsuFont.GetFont(size: 14, weight: FontWeight.Bold),
+                                Text = "show more".ToUpper(),
+                            },
+                            new ChevronIcon(),
+                        }
+                    },
+                    loading = new LoadingAnimation
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Size = new Vector2(20)
+                    },
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colors)
+            {
+                background.Colour = idleColour = colors.GreySeafoam;
+                hoveredColour = colors.GreySeafoamLight;
+            }
+
+            protected override bool OnHover(HoverEvent e)
+            {
+                background.FadeColour(hoveredColour, duration, Easing.OutQuint);
+                return base.OnHover(e);
+            }
+
+            protected override void OnHoverLost(HoverLostEvent e)
+            {
+                background.FadeColour(idleColour, duration, Easing.OutQuint);
+                base.OnHoverLost(e);
+            }
+
+            protected override bool OnClick(ClickEvent e)
+            {
+                Action.Invoke();
+                return base.OnClick(e);
+            }
+
+            private class ChevronIcon : SpriteIcon
```

Comment:
any reason you decided to make this specifically a nested subclass?

Reply:
just to avoid reduplication

Reply:
ah, didn't see it was used twice.